openslides_backend/action/actions/mediafile/upload.py

```diff
@@ -31,25 +32,71 @@
-    CreateActionWithDependencies,
-    CreateActionWithListOfSpeakersMixin,
-):
+class MediafileUploadAction(MediafilePermissionMixin, CreateAction):
     """
     Action to upload a mediafile.
     """
 
     model = Mediafile()
     schema = DefaultSchema(Mediafile()).get_create_schema(
-        required_properties=["title", "meeting_id", "filename"],
-        optional_properties=["access_group_ids", "parent_id"],
+        required_properties=["title", "owner_id", "filename"],
+        optional_properties=["token", "access_group_ids", "parent_id"],
         additional_required_fields={"file": {"type": "string"}},
     )
     permission = Permissions.Mediafile.CAN_MANAGE
 
-    dependencies = [ListOfSpeakersCreate]
+    @original_instances
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        tokens: List[Any] = []
+        for instance in action_data:
+            collection, _ = self.get_owner_data(instance)
+            if collection != "organization":
+                continue
+            tokens.append(instance.get("token"))
+            results = self.datastore.filter(
+                self.model.collection,
+                FilterOperator("token", "=", instance["token"]),
+                ["id"],
+            )
+            if len(results) == 0:
+                continue
+            elif len(results) == 1:
+                id = next(iter(results))
+                self.execute_other_action(MediafileDelete, [{"id": id}])
+            else:
+                text = f'Database corrupt: The resource token has to be unique, but there are {len(results)} tokens "{instance["token"]}".'
+                self.logger.error(text)
+                raise ActionException(text)
+        if len(tokens) != len(set(tokens)):
+            raise ActionException(
+                "It is not permitted to use the same token twice in a request."
+            )
+        return action_data
 
     def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        collection, _ = self.get_owner_data(instance)
+        if collection == "meeting":
+            return self.update_meeting_instance(instance)
+        elif collection == "organization":
+            return self.update_organization_instance(instance)
+        else:
+            return instance
+
+    def update_organization_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        instance["create_timestamp"] = round(time())
+        filename_ = instance.pop("filename")
+        file_ = instance.pop("file")
+        instance["mimetype"] = mimetypes.guess_type(filename_)[0]
+        if instance["mimetype"] is None:
+            raise ActionException(f"Cannot guess mimetype for {filename_}.")
+        decoded_file = base64.b64decode(file_)
+        instance["filesize"] = len(decoded_file)
+        id_ = instance["id"]
+        mimetype_ = instance["mimetype"]
+        self.media.upload_mediafile(file_, id_, mimetype_)
+        return instance
```

Comment:
You can combine this with the `update_meeting_instance`: Everything done here should also be done for the meeting case. `pop`ing the `filename` is not necessary and `pop`ing the `file` itself must also be done for the meeting case - it seems like you do not do this atm. Please also add a test for this if that is the case.

Reply:
Merge update_organization_instance and update_meeting_instance. Modify the code. File needs be popped and was popped in both methods.